Automattic/wp-calypso
# Add/app promo to login link

Before:
<img width="1119" alt="Screenshot 2023-10-17 at 4 39 38 PM" src="https://github.com/Automattic/wp-calypso/assets/115071/28aa0e54-939c-4c7c-82d5-f39e3a532c7b">

After:

<img width="887" alt="Screenshot 2023-10-20 at 3 21 51 PM" src="https://github.com/Automattic/wp-calypso/assets/115071/0e732cc5-ff76-4a91-85fc-37ea28eac19c">


iPhone
<img width="396" alt="Screenshot 2023-10-20 at 3 22 05 PM" src="https://github.com/Automattic/wp-calypso/assets/115071/d359c1c1-046f-4cb2-990f-045b7a5a25bc">



Android
<img width="417" alt="Screenshot 2023-10-20 at 3 23 02 PM" src="https://github.com/Automattic/wp-calypso/assets/115071/7b3acce7-f1db-4d67-a856-5063668bb819">



## Proposed Changes

* This Pr adds the app promo banner to the email me a login link pages. 


## Testing Instructions
* Load this PR. 
* Navigate to /log-in/link
* Notice the new banner. Make sure that it works as expected (responsive, loaded as an iPhone and Android) You should only see the app store button.
* Enter a valid email. Notice that the "Check your email" page also has the app banner.
* Visit /home/example.com notice that the admin bar looks and works as before. 
* Visit /me/get-apps notice that it works as before. 
* visit /devdocs/blocks/app-promo notice the app banner works as before. 

## Pre-merge Checklist
